### PR TITLE
Update template to support arm64

### PR DESCRIPTION
**Pull request by blackchoey**
1. Change to next generation python SDK and update sample code
2. Update docker images to support arm64

### CODE REVIEW

File: {{cookiecutter.module_name}}/Dockerfile.amd64 [hunk 1]
```diff
@@ -1,18 +1,10 @@
-FROM ubuntu:xenial
+FROM python:3.7-slim-buster
```
Comment (blackchoey) [inline {{cookiecutter.module_name}}/Dockerfile.amd64]: Python base image has no arch tag, so only be able to use the universal one

File: {{cookiecutter.module_name}}/main.py [hunk 2]
```diff
@@ -1,94 +1,74 @@
...
+from six.moves import input
+import threading
+from azure.iot.device.aio import IoTHubModuleClient
+
+async def main():
+    try:
+        print ( "\nPython %s\n" % sys.version )
+        print ( "IoT Hub Client for Python" )
+
+        print ( "The sample is now waiting for messages and will indefinitely. ")
```
Comment (adashen) [inline {{cookiecutter.module_name}}/main.py]: will indefinitely?

File: {{cookiecutter.module_name}}/main.py [hunk 3]
```diff
@@ -1,94 +1,74 @@
...
+        await user_finished
+
+        # Cancel listening
+        listeners.cancel()
+
+        # Finally, disconnect
+        await module_client.disconnect()
+
+    except Exception as e:
+        print ( "Unexpected error %s " % e )
```
Comment (adashen) [inline {{cookiecutter.module_name}}/main.py]: if exception, we didn't throw out the exception and exit?

Comment (blackchoey) [reply {{cookiecutter.module_name}}/main.py]: I'm following existing python template behavior. The module will exit after printed the error message. Do you mean we throw an exception here?

File: {{cookiecutter.module_name}}/main.py (hunk 2 above)
Comment (blackchoey) [reply {{cookiecutter.module_name}}/main.py]: Followed existing template. Will update the wording and move to latter place.

File: {{cookiecutter.module_name}}/main.py (hunk 3 above)
Comment (adashen) [reply {{cookiecutter.module_name}}/main.py]: If exit normally, that means if the outside restart policy is on-failure, it will not work

**Comment by blackchoey** [comment]
> We may also consider the filter module which will later be posted to python tutorial on Docs after GA.

The existing sample code of filter module on DOCS cannot be used with the new Python SDK, since the interface totally changed. I have created a task to figure out the new sample code.

**Pull request merged**